src/core/exec-invoke.c

```diff
@@ -555,8 +569,17 @@ static int setup_output(
                 return open_null_as(O_WRONLY, fileno);
 
         case EXEC_OUTPUT_TTY:
-                if (exec_input_is_terminal(i))
```

Comment:
this is unnecessary. The input is just reopened by us after all, so we know it's writable